services/sources/publisher_backlist_service.py

```diff
@@ -158,41 +159,76 @@ def get_records(
-                bucket = self.file_bucket if not record_permissions['requires_login'] else self.limited_file_bucket
-                s3_path = f'{self.title_prefix}/{record_metadata[SOURCE_FIELD][0]}/{file_name}'
-                s3_response = self.s3_manager.putObjectInBucket(file.getvalue(), s3_path, bucket)
-                
-                if not s3_response.get('ResponseMetadata').get('HTTPStatusCode') == 200:
-                    logger.error(f'Failed to upload file for {record_metadata.get("DRB_Record ID")} to S3')
-                    continue
-
-                s3_url = f'https://{bucket}.s3.amazonaws.com/{s3_path}'
                 
                 publisher_backlist_record = PublisherBacklistMapping(record_metadata)
                 publisher_backlist_record.applyMapping()
                 
-                webpub_flags=FileFlags(reader=True, limited_access=True)
-                self.add_has_part_mapping(s3_url, publisher_backlist_record.record, record_permissions['is_downloadable'], record_permissions['requires_login'])
-                self.s3_manager.store_pdf_manifest(publisher_backlist_record.record, self.file_bucket, flags=webpub_flags, path='/publisher_backlist')
+                try:
+                    file_url = self.download_file_from_location(record_metadata, record_permissions, publisher_backlist_record.record)
+                except Exception:
+                    logger.exception(f'Failed to download file for {record}')
+                    continue
                 
+                publisher_backlist_record.record.has_part.append(str(Part(
+                    index=1,
+                    url=file_url,
+                    source=publisher_backlist_record.record.source,
+                    file_type='application/pdf',
+                    flags=str(FileFlags(download=record_permissions['is_downloadable'], nypl_login=record_permissions['requires_login']))
+                )))
+                self.s3_manager.store_pdf_manifest(publisher_backlist_record.record, self.file_bucket, flags=FileFlags(reader=True, nypl_login=record_permissions['requires_login']), path='publisher_backlist')
                 mapped_records.append(publisher_backlist_record)
             except Exception:
                 logger.exception(f'Failed to process Publisher Backlist record: {record_metadata}')
         
         return mapped_records
+    
+    def download_file_from_location(self, record_metadata: dict, record_permissions: dict, record: Record) -> str:
+        file_location = record_metadata.get('DRB_File Location')
+        aws_file_bucket = os.environ.get("FILE_BUCKET", "drb-files-local")
+
+        if not file_location:
+            hath_identifier = next((identifier for identifier in record.identifiers if identifier.endswith('hathi')), None)
+
+            if hath_identifier is not None:
+                hathi_id = hath_identifier.split('|')[0]
+
+                s3 = boto3.client("s3")
```

Comment:
small thing I missed earlier here, but I think we should just reuse the `self.s3_manager.s3Client` instead of instantiating a fresh client on every call here

Reply:
That makes sense. i'll implement that.